Source/Meadow.Foundation.Peripherals/Sensors.Temperature.Mcp960x/Driver/Mcp960x.cs

```diff
@@ -0,0 +1,293 @@
+﻿using Meadow.Hardware;
+using Meadow.Peripherals.Sensors;
+using System;
+using System.Threading.Tasks;
+
+namespace Meadow.Foundation.Sensors.Temperature
+{
+    /// <summary>
+    /// Represents a Mcp960x Thermocouple sensor object
+    /// </summary>    
+    public abstract partial class Mcp960x : ByteCommsSensorBase<(Units.Temperature? TemperatureHot, Units.Temperature? TemperatureCold)>, ITemperatureSensor
+    {
+        /// <summary>
+        /// Raised when the Hot temperature value changes
+        /// </summary>
+        public event EventHandler<IChangeResult<Units.Temperature>> TemperatureUpdated = delegate { };
+
+        /// <summary>
+        /// Raised when the Hot temperature value changes
+        /// </summary>
+        public event EventHandler<IChangeResult<Units.Temperature>> TemperatureHotUpdated = delegate { };
+
+        /// <summary>
+        /// Raised when the Cold / ambient temperature value changes
+        /// </summary>
+        public event EventHandler<IChangeResult<Units.Temperature>> TemperatureColdUpdated = delegate { };
+
+        /// <summary>
+        /// The Hot Temperature value from the last reading
+        /// </summary>
+        public Units.Temperature? Temperature => TemperatureHot;
+
+        /// <summary>
+        /// The Hot Temperature value from the last reading
+        /// </summary>
+        public Units.Temperature? TemperatureHot { get; protected set; }
+
+        /// <summary>
+        /// The Cold Temperature value from the last reading
+        /// </summary>
+        public Units.Temperature? TemperatureCold { get; protected set; }
+
+        /// <summary>
+        /// Create a new Mcp960x object using the default configuration for the sensor
+        /// </summary>
+        /// <param name="i2cBus">The I2C bus</param>
+        /// <param name="address">I2C address of the sensor</param>
+        public Mcp960x(II2cBus i2cBus, byte address)
+            : base(i2cBus, address)
+        {
+            Peripheral.WriteRegister(DEVICECONFIG, 0x80);
+        }
+
+        /// <summary>
+        /// Update the Temperature property
+        /// </summary>
+        protected override Task<(Units.Temperature? TemperatureHot, Units.Temperature? TemperatureCold)> ReadSensor()
+        {
+            return Task.Run(() =>
+            {
+                (Units.Temperature? TemperatureHot, Units.Temperature? TemperatureCold) conditions;
+
+                conditions.TemperatureHot = new Units.Temperature(ReadTemperatureHot(), Units.Temperature.UnitType.Celsius);
+                conditions.TemperatureCold = new Units.Temperature(ReadTemperatureCold(), Units.Temperature.UnitType.Celsius);
+
+                return conditions;
+            });
+        }
+
+        /// <summary>
+        /// Reads the hot junction temperature from the MCP960x
+        /// </summary>
+        /// <returns>The hot junction temperature in degrees Celsius.</returns>
```

Comment:
Some comments have periods and some dont, want to consolidate while we're here?

Reply:
yeah I'll do a quick pass .... chat gpt adds em